runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/windmill/client/grpc/GrpcCommitWorkStream.java

```diff
@@ -187,13 +211,14 @@ protected void startThrottleTimer() {
     commitWorkThrottleTimer.start();
   }
 
-  private void flushInternal(Map<Long, PendingRequest> requests) {
+  private void flushInternal(Map<Long, PendingRequest> requests) throws InterruptedException {
```

Comment:
do we need `throws InterruptedException` here?

Reply:
we do since it is a checked exception thrown by one of the methods inside flushInternal
we handle it in flush()

wanted to make it clear the flushInternal is blocking()
if we get interrupted we interrupt the thread, but always clear the queue and queuedBytes (in the finally block)